internal/util/openapi/openapi.go

```diff
@@ -77,14 +88,89 @@ func ReadOpenAPISchemaFromDisk(path string) ([]byte, error) {
 
 func ConfigureOpenAPISchema(openAPISchema []byte) error {
 	openapi.SuppressBuiltInSchemaUse()
-
-	err := openapi.AddSchema(openAPISchema)
+	openAPISchema, err := addExtensionsToBuiltinTypes(openAPISchema)
 	if err != nil {
 		return err
 	}
-	// Also add make sure the Kustomize openAPI is always added regardless
-	// of where we got the Kubernetes openAPI schema.
-	// TODO: Refactor the openapi package in kyaml so we don't need to
-	// know the name of the kustomize asset here.
-	return openapi.AddSchema(kustomizationapi.MustAsset("kustomizationapi/swagger.json"))
+	if err := openapi.AddSchema(openAPISchema); err != nil {
+		return err
+	}
+	// Kustomize schema should always be added
+	return openapi.AddSchema(kustomizationapi.MustAsset(KustomizeAssetName))
+}
+
+// GetJSONSchema returns the JSON OpenAPI schema being used in kyaml
+func GetJSONSchema() ([]byte, error) {
+	schema := openapi.Schema()
+	if schema == nil {
+		return nil, nil
+	}
+	output, err := openapi.Schema().MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	var jsonSchema map[string]interface{}
+	if err := json.Unmarshal(output, &jsonSchema); err != nil {
+		return nil, err
+	}
+	if output, err = json.MarshalIndent(jsonSchema, "", "  "); err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func GetSchemaKRM() (*kyaml.RNode, error) {
+	schema, err := kyaml.Parse(`
+apiVersion: kpt.dev/v1
+kind: OpenAPI
+metadata:
+  name: openAPI
+  annotations:
+    config.kubernetes.io/path: "openapi_openAPI.yaml"
```

Comment:
This is just a fake name right?

Reply:
Yes, AFAICT this name doesn't matter since kpt will remove this resource from the output before writing anything out. We will need to have a regression test with a resource with the same file name.